cmd/create/machinepool/nodepool.go

```diff
@@ -31,9 +32,29 @@ func addNodePool(cmd *cobra.Command, clusterKey string, cluster *cmv1.Cluster, r
 		os.Exit(1)
 	}
 
-	// Hosted clusters create identifiers for NodePools, users don't interact directly with these resources
-	if cmd.Flags().Changed("name") {
-		r.Reporter.Errorf("Setting the `name` is not supported for hosted clusters")
+	// Machine pool name:
+	name := strings.Trim(args.name, " \t")
+	if name == "" && !interactive.Enabled() {
+		interactive.Enable()
+		r.Reporter.Infof("Enabling interactive mode")
+	}
+	if name == "" || interactive.Enabled() {
```

Comment:
this `name == ""` is unnecessary, as you're forcing interactive mode above. I would remove it.

Reply:
Yeah, I didn't realize.
I just reuse the same code used in the machinepool for classic rosa.

https://github.com/openshift/rosa/blob/master/cmd/create/machinepool/machinepool.go#L65

I prefer to add a little duplication here before jumping in a refactor for the common parameter (in favor to make this PR smaller).